### DISCUSSION TITLE
DSN example connection string for SQL Server -

### DISCUSSION THREAD

**Discussion by SakZeppelin** [burningalchemist/sql_exporter#541] [Q&A]
Hi

I'm new with this exporter trying to create custom metrics in order to create better alerts in Grafana but im struggling with the connection with my SQL Server Instance. 

I have a virtual machine in Azure where is installed my AlwaysOn Cluster in two VM and have an windows domain user with write permission in the sql instance.

Could you give me an example for DSN connection string for SQL Server?  (for a local sql user and also for an windows domain user)

My test DSN: sqlserver://sqluser:t492$JcG@vm-ag01:1433

Best regards

**Comment by burningalchemist** [comment]
Hi @SakZeppelin, I think you need to find out the SQL Server endpoint in Azure Web UI, the one you are using is a vm hostname, but it doesn't have the database instance.

Check the example here: https://github.com/burningalchemist/sql_exporter/blob/master/examples/azure-sql-mi/sql_exporter.yml (it also has example queries for Azure MSSQL)

**Reply by SakZeppelin** [reply]
Hi @burningalchemist  
Thanks for the answer.

The example is for a Azure Managed Instance, would shared it with a coworker who is working exactly with a managed instance.
In my case is an Azure SQL Server IAAS in a virtual machine 

I'm a bit confuse after see some documentations 

![image](https://github.com/burningalchemist/sql_exporter/assets/49209258/52fdac7c-7b2a-4294-9911-63dca46ce7b9)

https://learn.microsoft.com/en-us/azure/azure-sql/virtual-machines/windows/ways-to-connect-to-sql?view=azuresql

Or this another option: server=tcp:vm-ag01.myvirtualnetwork.prod;User ID=user;Password=yourpassword  ?>

Thanks in advance

**Reply by burningalchemist** [reply]
Hey @SakZeppelin, did you figure out the solution in the end? :)
